drivers/dma/dma_imx_rt.c

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright (c) 2020 NXP Semiconductor INC.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common part of DMA drivers for imx rt series.
+ */
+
+#include <errno.h>
+#include <soc.h>
+#include <init.h>
+#include <kernel.h>
+#include <drivers/dma.h>
+#include <drivers/clock_control.h>
+
+#include "dma_imx_rt.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dma_imx_rt, CONFIG_DMA_LOG_LEVEL);
+
+struct dma_imx_rt_config {
+	DMA_Type *base;
+	DMAMUX_Type *dmamux_base;
+	void (*irq_config_func)(struct device *dev);
+};
+
+static __aligned(32) edma_tcd_t
+	tcdpool[DT_INST_0_NXP_IMX_EDMA_DMA_CHANNELS][CONFIG_DMA_TCD_QUEUE_SIZE];
```

Comment:
why isn't this in `struct dma_imx_rt_data`?

Reply:
the TCD poll need special alignment 32 bytes address alignment, while the dma_data has no such requirement, just to keep the dma data compact. Any recommandation on how to use 32 bytes alignment in a structure? @MaureenHelm